dojo/tools/edgescan/parser.py

```diff
@@ -0,0 +1,78 @@
+import json
+
+from dojo.models import Endpoint, Finding
+from dojo.tools.edgescan.importer import EdgescanImporter
+
+ES_SEVERITIES = {1: "Info", 2: "Low", 3: "Medium", 4: "High", 5: "Critical"}
+SCANTYPE_EDGESCAN_API = 'Edgescan API Scan'
+SCANTYPE_EDGESCAN = 'Edgescan Scan'
+
+
+class EdgescanParser(object):
+    """
+    Import from Edgescan API or JSON file
+    """
+
+    def get_scan_types(self):
+        return [SCANTYPE_EDGESCAN, SCANTYPE_EDGESCAN_API]
+
+    def get_label_for_scan_types(self, scan_type):
+        return scan_type
+
+    def get_description_for_scan_types(self, scan_type):
+        if scan_type == SCANTYPE_EDGESCAN_API:
+            return "Edgescan findings can be imported by API."
+        return "Edgescan findings can be imported in JSON format."
+
+    def requires_file(self, scan_type):
+        return scan_type == SCANTYPE_EDGESCAN
+
+    def requires_tool_type(self, scan_type):
+        return scan_type == SCANTYPE_EDGESCAN_API
+
+    def get_findings(self, file, test):
+        try:
+            if file:
+                serialized_data = file.read()
+                try:
+                    data = json.loads(str(serialized_data, "utf-8"))
+                except:
+                    data = json.loads(serialized_data)
+            else:
+                data = EdgescanImporter().get_findings(test)
+        except:
+            raise Exception("Invalid details provided")
```

Comment:
Please don't try to manage by hand the encoding.

Just use `json.load(file)`
```suggestion
        if file:
            data = json.load(file)
        else:
            data = EdgescanImporter().get_findings(test)
```

Reply:
Fixed in 424bfaf.